client/patch-react.js

```diff
@@ -3,6 +3,15 @@
 
 import React from 'react'
 
+const getWritableProto = (obj, p) => {
+  const descr = Object.getOwnPropertyDescriptor(obj, p)
+  if (descr) {
+    return descr.writable ? obj : null
+  }
+  const proto = Object.getPrototypeOf(obj)
+  return proto ? getWritableProto(proto, p) : null
```

Comment:
This traverses the prototype chain, returning the prototype object on which the property is defined. `getOwnPropertyDescriptor` returns the descriptor for the immediate/own properties only.